src/clients/scopes/AddScopeDialog.tsx

```diff
@@ -50,55 +100,207 @@ export const AddScopeDialog = ({
     toggleDialog();
   };
 
+  const onFilterTypeDropdownToggle = () => {
+    setIsFilterTypeDropdownOpen(!isFilterTypeDropdownOpen);
+  };
+
+  const onProtocolTypeDropdownToggle = () => {
+    setIsProtocolTypeDropdownOpen(!isProtocolTypeDropdownOpen);
+  };
+
+  const onFilterTypeDropdownSelect = (filterType: string) => {
+    if (filterType === "Name") {
+      setFilterType(FilterType.Protocol);
+    }
+    if (filterType === "Protocol") {
+      setFilterType(FilterType.Name);
+    }
+    setIsFilterTypeDropdownOpen(!isFilterTypeDropdownOpen);
+  };
+
+  const onProtocolTypeDropdownSelect = (protocolType: string) => {
+    if (protocolType === "SAML") {
+      setProtocolType(ProtocolType.SAML);
+    }
+    if (protocolType === "OpenID Connect") {
+      setProtocolType(ProtocolType.OpenIDConnect);
+    }
+    if (protocolType === "All") {
+      setProtocolType(ProtocolType.All);
+    }
+    setIsProtocolTypeDropdownOpen(!isProtocolTypeDropdownOpen);
+  };
```

Comment:
Also here you can compare the value directly to the enum. 

```suggestion
  const onProtocolTypeDropdownSelect = (protocolType: string) => {
    if (protocolType === ProtocolType.SAML) {
      setProtocolType(ProtocolType.SAML);
    } else if (protocolType === ProtocolType.OpenIDConnect) {
      setProtocolType(ProtocolType.OpenIDConnect);
    } else if (protocolType === ProtocolType.All) {
      setProtocolType(ProtocolType.All);
    }

    setIsProtocolTypeDropdownOpen(!isProtocolTypeDropdownOpen);
  };
```